### PR TITLE
chore: update node version across the repo to 18

### PR DESCRIPTION
**Pull request by mrrajamanickam-coveo**
Cleanup and unify the nodejs versions used in different places in the mono repo. 

Currently almost all GH actions (except quantic) use node LTS (v18) via `.nvmrc`. This PR cleanups up references to node versions used elsewhere to match what is used in GH actions. 

TODO

- [ ] ~~`.github/actions/parse-version/action.yml`~~
  - https://json.schemastore.org/github-action.json: `"enum": ["node12", "node16"]` 

https://coveord.atlassian.net/browse/KIT-2443

### CODE REVIEW

**Comment by mrrajamanickam-coveo** [comment]
Test failures: 

- 2 test suites with couple of test failures
  -  recs-list.cypress.ts                       
  - smart-snippet.cypress.ts             

Given the nature of changes in this PR this does not seem concerning/relevant. 
Quantic test suites passed.

File: Jenkinsfile [hunk 1]
```diff
@@ -19,7 +19,7 @@ node('heavy && linux && docker') {
   }
 
   withEnv(['npm_config_cache=npm-cache', 'CI=true', 'NODE_OPTIONS=--max_old_space_size=8192']) {
-    withDockerContainer(image: 'node:16', args: '-u=root -e HOME=/tmp -e NPM_CONFIG_PREFIX=/tmp/.npm') {
+    withDockerContainer(image: 'node:18', args: '-u=root -e HOME=/tmp -e NPM_CONFIG_PREFIX=/tmp/.npm') {
```
Comment (mrrajamanickam-coveo) [inline Jenkinsfile:22]: Needs to be validated in the scheduled release after this PR is merged

**Review by louis-bompart — CHANGES_REQUESTED** [review]
Overall LGTM, watchout for the public package engine tho.

File: Jenkinsfile (hunk 1 above)
Comment (btaillon) [reply Jenkinsfile:22]: Eventually, I'm thinking we should switch to use [Volta](https://volta.sh/) instead. However, I recently tried (in my free time) to use Volta in Docker and it was a pain. It worked very well in GitHub Actions and on Windows though.

**Review by btaillon — APPROVED** [review]
Apart from the breaking changes that Louis pointed out, it's great!

**Review by mrrajamanickam-coveo — COMMENTED** [review]
Reverted node version updates in packages due to backwards compatibility concerns. 
Although practically since we don't test with the older node versions they are practically unsupported anyways :)

**Review by louis-bompart — APPROVED** [review]
lgtm

File: .github/actions/parse-version/action.yml [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
+    description: 'The minor version of Headless (e.g., 2.12)'
   atomic-major:
-    description: "The major version of Atomic (e.g., 2)"
+    description: 'The major version of Atomic (e.g., 2)'
   atomic-minor:
-    description: "The minor version of Atomic (e.g., 2.12)"
+    description: 'The minor version of Atomic (e.g., 2.12)'
 runs:
+  # TODO(KIT-2443): Update to node LTS, currently yaml schema allows only: 12, 16
+  #   https://json.schemastore.org/github-action.json: "enum": ["node12", "node16"]
```
Comment (louis-bompart) [inline .github/actions/parse-version/action.yml:14]: @mrrajamanickam-coveo : https://github.com/orgs/community/discussions/53217#discussioncomment-6758913
We should probably jump straight to 20 when available, as 16 is croaking in 3 weeks.

**Review by louis-bompart — APPROVED** [review]

File: .github/actions/parse-version/action.yml (hunk 2 above)
Comment (mrrajamanickam-coveo) [reply .github/actions/parse-version/action.yml:14]: 👍 Wish they also had a `LTS` tag to use

**Comment by mrrajamanickam-coveo** [comment]
CI test failure seems flaky/unrelated - proceeding to merge
```
  53 passing (2m)
  1 failing

  1) Smart Snippet Test Suites
       with no heading level
         should have links to the source:
     AssertionError: Timed out retrying after 4000ms: Expected to find element: `[part="source-url"]`, but never found it. Queried from element: <document>
      at Context.eval (webpack:///./cypress/e2e/smart-snippet.cypress.ts:55:40)
```

**Pull request merged**